tool/smarts/src/test/java/org/openscience/cdk/smirks/SmirksTest.java

```diff
@@ -761,4 +788,235 @@ void testDielsAlder() throws Exception {
+        final String[] expected = new String[] {"C1(=CC=NC=C1)O", "C1=C(C=NC=C1)O", "C1=CC(=NC=C1)O", "C1=CC=NC(=C1)O", "C1=CC=NC=C1O"};
+        assertTransform(smiles, smirks, "C1(=C(C(=NC(=C1O)O)O)O)O");
+        assertTransform(smiles, smirks, expected, Transform.Mode.Unique);
+        assertTransform(smiles, smirks, expected, Transform.Mode.All);
+    }
+
+    @Test
+    void testAmide_separateComponentsMultipleCombinations() throws Exception {
+        final String smiles = "NCC=C.NC1=CC=CC=C1.OC(=O)C1CCCOC1";
+        final String smirks = "[O:1]=[C:2][O:3].[NH2:4]>>[O:1]=[C:2][NH:4]";
+        final String[] expected = new String[] {"N(CC=C)C(=O)C1CCCOC1.NC1=CC=CC=C1", "NCC=C.N(C1=CC=CC=C1)C(=O)C2CCCOC2"};
+        assertTransform(smiles, smirks, "N(CC=C)C(=O)C1CCCOC1.NC1=CC=CC=C1");
+        assertTransform(smiles, smirks, expected, Transform.Mode.Unique);
+        assertTransform(smiles, smirks, expected, Transform.Mode.All);
+    }
+
+    @Test
+    void testAddingCarbon_infiniteLoopPrevention() throws Exception {
+        // the transformed molecule also matches the LHS of the smirks, assert that there is no infinite loop of matching
+        final String smiles = "CCCCN";
+        final String smirks = "[C:1][N:2]>>[C:1]C[N:2]";
+        final String[] expected = new String[] {"CCCCCN"};
+        assertTransform(smiles, smirks, "CCCCCN");
+        assertTransform(smiles, smirks, expected, Transform.Mode.Unique);
+        assertTransform(smiles, smirks, expected, Transform.Mode.All);
+    }
+
+    @Test
+    void testRingFormation_1() throws Exception {
+        // the transformed molecule also matches the LHS of the smirks, assert that there is no infinite loop of matching
+        final String smiles = "OC1NCCC(F)N1";
+        final String smirks = "[O:1][C:2][N:3][C:4]>>[OH0:1]1[C:2][NH0:3]([C:4])CC1";
+        final String[] expected = new String[] {"O1C2N(CCC(F)N2)CC1", "O1C2NCCC(F)N2CC1"};
+        assertTransform(smiles, smirks, "O1C2N(CCC(F)N2)CC1");
+        assertTransform(smiles, smirks, expected, Transform.Mode.Unique);
+        assertTransform(smiles, smirks, expected, Transform.Mode.All);
+    }
+
+    // the mapped Br atom is present on the LHS only and gets removed when applying the transform
+    @Test
+    void testMappedAndUnmappedAtoms_1() throws Exception {
+        final String smiles = "CCC(Br)=O.CCN";
+        final String smirks = "[C:1]C([Br:2])=[O:3].[C:11][N:10]>>[C:1]C(=[O:3])[NH1:10][C:11]";
+        final String expected = "CCC(=O)NCC";
+
+        assertTransform(smiles, smirks, "CCC(=O)NCC");
+        assertTransform(smiles, smirks, new String[] {expected}, Transform.Mode.Unique);
+        assertTransform(smiles, smirks, new String[] {expected}, Transform.Mode.All);
+    }
+
+    @Disabled("the unmapped Br atom is present on the LHS only and gets removed when applying the transform; I would have expected the Br atom to not be removed as it is not mapped...?")
```

Comment:
Not correct, unmapped atoms are removed. The default normally is you error on unpaired mapped atoms. Daylight diss-allowed them complete.